src/file.ts

```diff
@@ -1802,25 +1804,32 @@ class File extends ServiceObject<File> {
       const configDir = xdgBasedir.config || os.tmpdir();
 
       fs.access(configDir, fs.constants.W_OK, err => {
-        if (err) {
-          if (options.resumable) {
-            const error = new ResumableUploadError(
-              [
-                'A resumable upload could not be performed. The directory,',
-                `${configDir}, is not writable. You may try another upload,`,
-                'this time setting `options.resumable` to `false`.',
-              ].join(' ')
-            );
-            stream.destroy(error);
-            return;
+        const maybeCreateFolder = async () => {
+          if (err) {
+            try {
+              await mkDirAsync(configDir, {mode: 0o0700});
+            } catch (mkDirErr) {
+              if (options.resumable) {
+                const error = new ResumableUploadError(
+                  [
+                    'A resumable upload could not be performed. The directory,',
```

Comment:
I think there are two causes of failures now:

- The directory doesn't exist, and can't be created
- The directory exists, but cannot be written to

Should we generalize the error message or maybe do a separate message for each?

Reply:
Great callout! I've created a separate issue to track this [here](https://github.com/googleapis/nodejs-storage/issues/1431).